include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -226,19 +226,46 @@ namespace internal
 
     // helper function
     const auto sort_constraints = [&]() {
+      if (locally_relevant_constraints.empty())
+        return;
+
       std::sort(locally_relevant_constraints.begin(),
                 locally_relevant_constraints.end(),
                 [](const auto &a, const auto &b) {
                   return std::get<0>(a) < std::get<0>(b);
                 });
 
-      locally_relevant_constraints.erase(
-        std::unique(locally_relevant_constraints.begin(),
-                    locally_relevant_constraints.end(),
-                    [](const auto &a, const auto &b) {
-                      return std::get<0>(a) == std::get<0>(b);
-                    }),
-        locally_relevant_constraints.end());
+      auto read_ptr  = locally_relevant_constraints.begin();
+      auto write_ptr = locally_relevant_constraints.begin();
+      while (++read_ptr != locally_relevant_constraints.end())
+        {
+          if (std::get<0>(*read_ptr) != std::get<0>(*write_ptr))
+            {
+              if (++write_ptr != read_ptr)
+                *write_ptr = std::move(*read_ptr);
+            }
+          else
+            {
+              auto &      a = *write_ptr;
+              const auto &b = *read_ptr;
+              Assert(std::get<0>(a) == std::get<0>(b), ExcInternalError());
+              if (std::get<1>(a) != std::get<1>(b))
+                std::get<1>(a) += std::get<1>(b);
```

Comment:
This is adding the inhomogeneous values when they are different, right? Why is the sum correct? Are you assuming that one of them is 0 here? What if both are nonzero and only differ by eps (assume we use some computation for boundary values that is not 100% reproducible/exact).

Reply:
Actually, I do not have any inhomogeneous constraints in my use case. What do you think would be the consistent strategy here? Checking the difference with a certain tolerance, and then adding up?